### PR TITLE
Add user management APIs

### PR DESCRIPTION
**Pull request by shilgapira**
## Related Issues
Resolves https://github.com/descope/etc/issues/1499

## Related PRs
https://github.com/descope/managementservice/pull/207

## Description
Add user management APIs for updating specific fields

## Must
- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: lib/management/user.ts [hunk 1]
```diff
@@ -83,6 +83,117 @@ const withUser = (sdk: CoreSdk, managementKey?: string) => ({
       ),
       (data) => data.users,
     ),
+  activate: (loginId: string): Promise<SdkResponse<UserResponse>> =>
+    transformResponse<SingleUserResponse, UserResponse>(
```
Comment (nirgur) [inline lib/management/user.ts:87]: What's the reason for specifying the types twice?
(on the function return type & on the transformResponse types)

File: lib/management/user.test.ts [hunk 2]
```diff
@@ -210,4 +210,341 @@ describe('Management User', () => {
...
+  describe('activate', () => {
+    it('should send the correct request and receive correct response', async () => {
+      const httpResponse = {
+        ok: true,
+        json: () => mockMgmtUserResponse,
+        clone: () => ({
+          json: () => Promise.resolve(mockMgmtUserResponse),
+        }),
+        status: 200,
+      };
```
Comment (nirgur) [inline lib/management/user.test.ts:223]: I see this part repeats in every test
maybe we can extract it into a utility function?

**Review by nirgur — COMMENTED** [review]
Looks good 👍 
Added 2 comments

**Review by itaihanski — APPROVED** [review]
Looks good 👍  I suggest we add function documentation on the new functions

**Pull request merged**